lib/api/src/sys/mem_access.rs

```diff
@@ -146,9 +146,9 @@ impl<'a, T: ValueType> fmt::Debug for WasmRef<'a, T> {
 ///
 /// This wrapper safely handles concurrent modifications of the data by another
 /// thread.
-#[derive(Clone, Copy)]
+#[derive(Clone)]
 pub struct WasmSlice<'a, T: ValueType> {
-    buffer: MemoryBuffer<'a>,
+    view: MemoryGuard<'a>,
```

Comment:
I'm not super sure about this changes. Once we have two kinds of Buffer (non shared and shared), I think this might need to vary more

Reply:
the threading extension proposes a new Shared Memory object. We will need two kinds of buffer, shared and not shared.

Reply:
Now implemented it with Shared and Owned memory objects closer to the threading proposal